src/bo4e_generator/sqlparser.py

```diff
@@ -0,0 +1,235 @@
+                    f'List["{reference_name}"] ='
+                    f' Relationship(back_populates="{class_name.lower()}_link", '
+                    f"link_model={class_name}{reference_name}Link)"
+                )
+                add_fields[reference_name][f"{class_name.lower()}_link"] = (
+                    f'List["{class_name}"] ='
+                    f' Relationship(back_populates="{field_name}", '
+                    f"link_model={class_name}{reference_name}Link)"
+                )
+                add_imports[class_name + "ADD"][f"{class_name}{reference_name}Link)"] = "Link"
+                add_imports[reference_name + "ADD"][f"{class_name}{reference_name}Link)"] = "Link"
+            else:
+                add_fields[class_name][f"{field_name}_id"] = (
+                    "uuid_pkg.UUID "
+                    + is_optional
+                    + f' = Field(default=None, foreign_key="{reference_name.lower()}.{reference_name.lower()}_sqlid")'
+                )
+                add_fields[class_name][f"{field_name}"] = (
+                    f'"{reference_name}" ='
+                    f' Relationship(back_populates="{class_name.lower()}_{field_name}",'
+                    f' sa_relationship_kwargs= {{ "foreign_keys":"[{class_name}.{field_name}_id]" }})'
+                )
+                # add_imports[class_name + "ADD"]["Optional"] = "typing"
+
+                add_fields[reference_name][f"{class_name.lower()}_{field_name}"] = (
+                    f'List["{class_name}"] = Relationship(back_populates="{field_name}",'
+                    f"sa_relationship_kwargs="
+                    f'{{"primaryjoin":'
+                    f' "{class_name}.{field_name}_id=={reference_name}.{reference_name.lower()}_sqlid",'
+                    f' "lazy": "joined" }})'
+                )
+            # add_relation_import
+            add_imports[class_name][
+                reference_name
+            ] = f"{namespace[reference_name].pkg}.{namespace[reference_name].module_name}"
+            add_imports[reference_name][class_name] = f"{namespace[class_name].pkg}.{namespace[class_name].module_name}"
+
+    return add_fields, add_imports
+
+
+def write_many_many_links(links: dict[str, str]) -> str:
+    """
+    use template to write many-to-many link classes to many.py file
+    """
+    template_path = Path(__file__).resolve().parent / Path("custom_templates")
+    environment = Environment(loader=FileSystemLoader(template_path))
+    template = environment.get_template("ManyLinks.jinja2")
+    python_code = template.render({"class": links})
+    python_code = black.format_str(python_code, mode=black.Mode())
+    python_code = isort.code(python_code)
+    return python_code
+
+
+def isort_directory(code: str) -> str:
+    """
+    perform isort on the output dir
+    """
+    code = black.format_str(code, mode=black.Mode())
+    return isort.code(code, known_local_folder=["borm"])
```

Comment:
Nice 👍 Aber warum hast du das in Zeile 225/226 nochmal stehen? Kannst du da nicht diese Funktion einfach aufrufen?
Btw, falls noch nicht geschehen, das kannst du gerne direkt in der main-Funktion immer auf jede Datei einfach am Ende performen - me like 👍